x-pack/filebeat/module/misp/threat/config/pipeline.js

```diff
@@ -105,71 +114,87 @@ var threat = (function () {
-                attackPattern = '[' + 'github.username = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'github:username = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'github.username: ' + '"' + v + '"';
                 break;
             case "hostname":
-                attackPattern = '[' + 'source.domain = ' + '\'' + v + '\'' + ' OR destination.domain = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'source:domain = ' + '\'' + v + '\'' + ' OR destination:domain = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'source.domain: ' + '"' + v + '"' + ' OR destination.domain: ' + '"' + v + '"';
                 evt.Put("source.domain", v);
                 evt.Put("destination.domain", v);
                 break;
             case "ip-dst":
                 ip = v.split("/")[0];
-                attackPattern = '[destination.ip = ' + '\'' + ip + '\'' + ']';
+                attackPattern = '[destination:ip = ' + '\'' + ip + '\'' + ']';
+                attackPatternKQL = 'destination.ip: ' + '"' + ip + '"';
                 evt.Put("destination.ip", ip);
                 break;
             case "ip-dst|port":
                 arr = v.split("|");
                 if (arr.length == 2) {
-                  attackPattern = '[destination.ip = ' + '\'' + arr[0] + '\'' + ' AND destination.port = ' + '\'' + arr[1] + '\'' + ']';
+                  attackPattern = '[destination:ip = ' + '\'' + arr[0] + '\'' + ' AND destination:port = ' + '\'' + arr[1] + '\'' + ']';
+                  attackPatternKQL = 'destination.ip: ' + '"' + arr[0] + '"' + ' AND destination.port: ' + arr[1];
                   evt.Put("destination.ip", arr[0]);
+                  evt.Put("destination.port", arr[1]);
                 }
                 break;
             case "ip-src":
                 ip = v.split("/")[0];
-                attackPattern = '[' + 'source.ip = ' + '\'' + ip + '\'' + ']';
+                attackPattern = '[' + 'source:ip = ' + '\'' + ip + '\'' + ']';
+                attackPatternKQL = 'source.ip: ' + '"' + ip + '"';
                 evt.Put("source.ip", ip);
                 break;
             case "link":
-                attackPattern = '[' + 'url.full = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'url:full = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'url.full: ' + '"' + v + '"';
                 evt.Put("url.full", v);
                 break;
             case "md5":
-                attackPattern = '[' + 'file.hash.md5 = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'file:hash:md5 = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'file.hash.md5: ' + '"' + v + '"';
                 evt.Put("file.hash.md5", v);
                 break;
             case 'regkey':
                 attackPattern = '[' + 'regkey = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'regkey: ' + '"' + v + '"';
```

Comment:
Registry fields haven't been formalized in ECS yet so this one will change in the future.

Reply:
Yes